ios/FluentUI/Vnext/IndeterminateProgressBar/IndeterminateProgressBar.swift

```diff
@@ -0,0 +1,149 @@
+}
+
+/// View that represents the Indeterminate Progress Bar
+public struct IndeterminateProgressBar: View {
+    @Environment(\.theme) var theme: FluentUIStyle
+    @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+    @ObservedObject var tokens: MSFIndeterminateProgressBarTokens
+    @ObservedObject var state: MSFIndeterminateProgressBarStateImpl
+    @State var startPoint: UnitPoint = Constants.initialStartPoint
+    @State var endPoint: UnitPoint = Constants.initialEndPoint
+
+    /// Creates the Indeterminate Progress Bar.
+    public init() {
+        let state = MSFIndeterminateProgressBarStateImpl()
+        self.state = state
+        self.tokens = state.tokens
+    }
+
+    public var body: some View {
+        let height = tokens.height
+        let gradientColor = Color(tokens.gradientColor)
+        let backgroundColor = Color(tokens.backgroundColor)
+
+        Rectangle()
+            .fill(LinearGradient(gradient: Gradient(colors: [backgroundColor, gradientColor, backgroundColor]),
+                                 startPoint: startPoint,
+                                 endPoint: endPoint))
+            .frame(maxWidth: .infinity,
+                   minHeight: height,
+                   idealHeight: height,
+                   maxHeight: height,
+                   alignment: .center)
+            .background(backgroundColor)
+            .modifyIf(state.isAnimating, { view in
+                view
+                    .onAppear {
+                        startAnimation()
+                    }
+                    .onDisappear {
+                        stopAnimation()
+                    }
+            })
+            .modifyIf(!state.isAnimating) { view in
+                view
+                    .onAppear {
+                       stopAnimation()
+                    }
+            }
+            .modifyIf(!state.isAnimating && state.hidesWhenStopped, { view in
+                view.hidden()
+            })
+            .designTokens(tokens,
+                          from: theme,
+                          with: windowProvider)
+    }
+
+    private func startAnimation() {
+        withAnimation(Animation.linear(duration: Constants.animationDuration)
+                                .repeatForever(autoreverses: false)) {
+            startPoint = Constants.finalStartPoint
+            endPoint = Constants.finalEndPoint
+        }
+    }
+
+    private func stopAnimation() {
+        withAnimation(Animation.linear(duration: 0)) {
+            startPoint = Constants.initialStartPoint
+            endPoint = Constants.initialEndPoint
+        }
+    }
+
+    private struct Constants {
+        static let animationDuration: Double = 1.75
+        static let initialStartPoint = UnitPoint(x: -1, y: 0.5)
+        static let initialEndPoint = UnitPoint(x: 0, y: 0.5)
+        static let finalStartPoint = UnitPoint(x: 1, y: 0.5)
+        static let finalEndPoint = UnitPoint(x: 2, y: 0.5)
```

Comment:
do these points swap for RTL?

Reply:
Good point. Although they have .leading and .trailing properties, they're not interpreted when an RTL language is set.
Is this the behavior we want?

https://user-images.githubusercontent.com/68076145/134084122-369939a9-5d42-45f8-9d66-e96ab3d1464a.mov